### PR TITLE
Including PartitionKeyDefinitionVersion

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Including PartitionKeyDefinitionVersion into containersettings

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs [hunk 1]
```diff
@@ -85,6 +85,27 @@ public CosmosContainerSettings(string id, string partitionKeyPath)
             ValidateRequiredProperties();
         }
 
+        /// <summary>
+        /// Gets the Partitioning scheme version used
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosContainerSettings.cs]: As is it public , some more documentation would be helpful ?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 2]
```diff
@@ -104,6 +105,22 @@ public async Task PartitionedCRUDTest()
...
+            string partitionKeyPath = "/users";
+
+            CosmosContainerSettings settings = new CosmosContainerSettings(containerName, partitionKeyPath);
+            settings.PartitionKeyDefinitionVersion = Cosmos.PartitionKeyDefinitionVersion.V1;
+
+            CosmosContainerResponse cosmosContainerResponse = await this.cosmosDatabase.Containers.CreateContainerAsync(settings);
+
+            Assert.AreEqual(HttpStatusCode.Created, cosmosContainerResponse.StatusCode);
+
+            Assert.AreEqual(PartitionKeyDefinitionVersion.V1, cosmosContainerResponse.Resource.PartitionKeyDefinitionVersion);
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Is this test passing with PartitionKeyDefinitionVersion.V1 without specifying Cosmos.PartitionKeyDefinitionVersion.V1  ?

Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Anyway good to mention which one Document or Cosmos

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs [hunk 3]
```diff
@@ -96,6 +96,7 @@ public async Task PartitionedCRUDTest()
             containerResponse = await cosmosContainer.ReadAsync();
             Assert.AreEqual(HttpStatusCode.OK, containerResponse.StatusCode);
             Assert.AreEqual(containerName, containerResponse.Resource.Id);
+            Assert.AreEqual(PartitionKeyDefinitionVersion.V2, containerResponse.Resource.PartitionKeyDefinitionVersion);
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Same here Cosmos.PartitionKeyDefinitionVersion.V1 ?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs (hunk 2 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs]: Both version values are same. One if an internal contract in direct other is SDK public contract. (Split brain :-( )

Also there is a tests to assert that the values are same between as well.

**Pull request merged**